### PR TITLE
Add json APIs to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/cudf/source/user_guide/api_docs/pylibcudf/strings/index.rst
Comment (davidwendt) [inline docs/cudf/source/user_guide/api_docs/pylibcudf/strings/index.rst:15]: The `get_json_object` and related APIs are not part of the `cudf::strings` namespace in libcudf.
https://github.com/rapidsai/cudf/blob/branch-24.12/cpp/include/cudf/json/json.hpp
Should pylibcudf match libcudf more closely than cudf?

Comment (Matt711) [reply docs/cudf/source/user_guide/api_docs/pylibcudf/strings/index.rst:15]: Yes pylibcudf should match libcudf more closely than cudf python. So, I think we should move `json.pxd` and `json.pyx` out of strings.

But a broader point: I think pylibcudf is first and foremost ["python bindings for libcudf"](https://docs.rapids.ai/api/cudf/stable/developer_guide/pylibcudf/#pylibcudf), so we should match libcudf as closely as possible. But pylibcudf is a python library so I think it makes sense over time to make it more "pythonic" (eg. easier to use with other python libraries).
cc. @vyasr

Comment (mroeschke) [reply docs/cudf/source/user_guide/api_docs/pylibcudf/strings/index.rst]: Thanks for the catch. I moved this out of the `string` namespace.

Comment (Matt711) [reply docs/cudf/source/user_guide/api_docs/pylibcudf/strings/index.rst]: > Or were you just responding to David's question with a statement that while pylibcudf APIs should closely match libcudf APIs, they may not match exactly if there is a more Pythonic choice for the pylibcudf API that still faithfully reflects the semantics of the C++ functions?

That's right. I realized I didn't add "Is that right?" before I @'d you because I wanted to get your thoughts.

File: python/pylibcudf/pyproject.toml
Comment (jameslamb) [inline python/pylibcudf/pyproject.toml:101]: According to https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html, `--import-mode=importlib` was added in `pytest` 6.0.

So we probably need a floor here:

https://github.com/rapidsai/cudf/blob/7173b52fce25937bb69e22a083a5de4655078fa1/dependencies.yaml#L749

like:

```text
- pytest>=6.0,<8
```

**Review by jameslamb — APPROVED** [review]
Giving you a `packaging-codeowners` approval. Left one packaging-related comment for your consideration. I don't need to re-review unless the packaging-related stuff changes. Will defer to other reviewers on the API changes in this PR.

File: python/pylibcudf/pyproject.toml
Comment (vyasr) [reply python/pylibcudf/pyproject.toml:101]: This is technically true, but I think we have other reasons why we wouldn't be compatible with lower pytest versions either and we're just generally underspecified. I don't think that we need to address this here and should instead look into this as part of https://github.com/rapidsai/build-planning/issues/105.

**Review by vyasr — APPROVED** [review]

**Pull request merged**
